hack/e2e.sh

```diff
@@ -101,20 +101,29 @@ function kubectl_get_jsonpath() {
   fi
 }
 
+function claim_ip() {
+  IPCLAIM_NAME="$1"
+  export IPCLAIM_NAME
+  envsubst < "${REPO_ROOT}/hack/ipclaim-template.yaml" | kubectl --kubeconfig="${KUBECONFIG}" create -f - 1>&2
+  IPADDRESS_NAME=$(kubectl_get_jsonpath ipclaim "${IPCLAIM_NAME}" '{@.status.address.name}')
+  kubectl --kubeconfig="${KUBECONFIG}" get "$(append_api_group ipaddresses)" "${IPADDRESS_NAME}" -o=jsonpath='{@.spec.address}'
+}
+
+export KUBECONFIG="/root/ipam-conf/capv-services.conf"
+
+make envsubst
+
 # Retrieve an IP to be used as the kube-vip IP
-KUBECONFIG="/root/ipam-conf/capv-services.conf"
-IPCLAIM_NAME="ip-claim-$(openssl rand -hex 20)"
-sed "s/IPCLAIM_NAME/${IPCLAIM_NAME}/" "${REPO_ROOT}/hack/ipclaim-template.yaml" | kubectl --kubeconfig=${KUBECONFIG} create -f -
-IPADDRESS_NAME=$(kubectl_get_jsonpath ipclaim "${IPCLAIM_NAME}" '{@.status.address.name}')
-CONTROL_PLANE_ENDPOINT_IP=$(kubectl --kubeconfig=${KUBECONFIG} get "$(append_api_group ipaddresses)" "${IPADDRESS_NAME}" -o=jsonpath='{@.spec.address}')
+CONTROL_PLANE_IPCLAIM_NAME="ip-claim-$(openssl rand -hex 20)"
+export CONTROL_PLANE_IPCLAIM_NAME
+CONTROL_PLANE_ENDPOINT_IP=$(claim_ip "${CONTROL_PLANE_IPCLAIM_NAME}")
 export CONTROL_PLANE_ENDPOINT_IP
 echo "Acquired Control Plane IP: $CONTROL_PLANE_ENDPOINT_IP"
 
 # Retrieve an IP to be used for the workload cluster in v1a3/v1a4 -> v1b1 upgrade tests
-WORKLOAD_IPCLAIM_NAME="workload-ip-claim-$(openssl rand -hex 20)"
-sed "s/IPCLAIM_NAME/${WORKLOAD_IPCLAIM_NAME}/" "${REPO_ROOT}/hack/ipclaim-template.yaml" | kubectl --kubeconfig=${KUBECONFIG} create -f -
-WORKLOAD_IPADDRESS_NAME=$(kubectl_get_jsonpath ipclaim "${WORKLOAD_IPCLAIM_NAME}" '{@.status.address.name}')
-WORKLOAD_CONTROL_PLANE_ENDPOINT_IP=$(kubectl --kubeconfig=${KUBECONFIG} get "$(append_api_group ipaddresses)" "${WORKLOAD_IPADDRESS_NAME}" -o=jsonpath='{@.spec.address}')
+WORKLOAD_PLANE_IPCLAIM_NAME="ip-claim-$(openssl rand -hex 20)"
```

Comment:
should we still add workload in some part of the name? Just makes it easier to distinguish later